torch_geometric/nn/conv/gmn_conv.py

```diff
@@ -0,0 +1,150 @@
+            :obj:`"var"` and :obj:`"std"`.
+        scalers (List[str]): Set of scaling function identifiers, namely
+            :obj:`"identity"`, :obj:`"amplification"`,
+            :obj:`"attenuation"`, :obj:`"linear"` and
+            :obj:`"inverse_linear"`.
+        deg (torch.Tensor): Histogram of in-degrees of nodes in the training
+            set, used by scalers to normalize.
+        edge_dim (int, optional): Edge feature dimensionality (in case
+            there are any). (default :obj:`None`)
+        towers (int, optional): Number of towers (default: :obj:`1`).
+        post_layers (int, optional): Number of transformation layers after
+            aggregation (default: :obj:`1`).
+        divide_input (bool, optional): Whether the input features should
+            be split between towers or not (default: :obj:`False`).
+        act (str or callable, optional): Pre- and post-layer activation
+            function to use. (default: :obj:`"relu"`)
+        act_kwargs (Dict[str, Any], optional): Arguments passed to the
+            respective activation function defined by :obj:`act`.
+            (default: :obj:`None`)
+        train_norm (bool, optional): Whether normalization parameters
+            are trainable. (default: :obj:`False`)
+        **kwargs (optional): Additional arguments of
+            :class:`torch_geometric.nn.conv.MessagePassing`.
+
+    Shapes:
+        - **input:**
+          node features :math:`(|\mathcal{V}|, F_{in})`,
+          edge indices :math:`(2, |\mathcal{E}|)`,
+          edge features :math:`(|\mathcal{E}|, D)` *(optional)*
+        - **output:** node features :math:`(|\mathcal{V}|, F_{out})`
+    """
+    def __init__(self, in_channels: int, 
+                 out_channels: int,
+                 aggregators: List[str], 
+                 scalers: List[str], 
+                 deg: Tensor,
+                 edge_dim: Optional[int] = None, 
+                 towers: int = 1,
+                 post_layers: int = 1,  
+                 divide_input: bool = False, 
+                 act: Union[str, Callable, None] = "relu",
+                 act_kwargs: Optional[Dict[str, Any]] = None,
+                 **kwargs):
+
+        super().__init__(in_channels, out_channels, aggregators, scalers, deg, 
+                         edge_dim, towers, divide_input, **kwargs)
+
+        if divide_input:
+            assert in_channels % towers == 0
+        assert out_channels % towers == 0
+
+        self.in_channels = in_channels
+        self.out_channels = out_channels
+        self.aggregators = aggregators
+        self.scalers = scalers
+        self.edge_dim = edge_dim
+        self.towers = towers
+        self.divide_input = divide_input
+
+        self.F_in = in_channels // towers if divide_input else in_channels
+        self.F_out = self.out_channels // towers
+
+        if self.edge_dim is not None:
+            self.edge_encoder = Linear(edge_dim, self.F_in)
```

Comment:
These are already defined with line 104, you should be able to just reuse them ? https://github.com/pyg-team/pytorch_geometric/blob/89f4873b411371880fb64344615ee9babef67d5e/torch_geometric/nn/conv/pna_conv.py#L112-L126

Reply:
Certainly, that approach works. I'll make the updates as you've suggested.